packages/styled/types/base.d.ts

```diff
@@ -77,10 +77,18 @@ export interface CreateStyledComponent<
       >
     >
   ): StyledComponent<ComponentProps & AdditionalProps, SpecificComponentProps>
+
+  (
```

Comment:
Could you explain how this has fixed the issue? I don't quite follow this. It's also an overload "in the middle" of listed overloads - is it intentionally at this particular position?

Reply:
Yeah sure

Overloads are selected from top to bottom. The issue was that the props of the second interpolated component was being inferred as AdditionalProps.

By introducing an overload without AdditionalProps it will be selected if it matches, which happens when it's not specified. Basically the overload removes the ability for AdditionalProps to be inferred.

So for example, this will work:

```
const Input4 = styled.input<AdditionalTest>`
  & + ${Label}: ${props => ({
    marginLeft: props.left
  })}
`
```

but this will not:

```
interface AdditionalTest {
  left: string
}
const Input4 = styled.input`
  & + ${Label}: ${(props: AdditionalTest) => ({
    marginLeft: props.left
  })}
`
```